src/registry/mod.rs

```
@@ -128,24 +129,33 @@ impl Registry {
         Ok(m)
     }
 
-    pub async fn push(&self, policy: &[u8], url: &str, sources: &Option<Sources>) -> Result<()> {
+    /// Fetch the manifest's digest of the OCI object referenced by the given url.
+    /// The url is expected to be in the "registry://" format.
+    pub async fn manifest_digest(&self, url: &str, sources: Option<&Sources>) -> Result<String> {
+        let url = Url::parse(url).map_err(|_| anyhow!("invalid URL: {}", url))?;
+        let reference =
+            Reference::from_str(url.as_ref().strip_prefix("registry://").unwrap_or_default())?;
```

I think we want to bubble the error up as well here, falling back to the default would end up in `Reference::from_str(String::default))` what could lead to errors hard to understand on the user side.

---

Good point, I have to admit I copied and pasted this snippet from other parts of the code. I've changed it now, can you take another look?